File: src/RNetPi.Core/Packets/PacketS2CZoneName.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x04
/// Zone Name
/// Sends a zone's current name
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (String) Zone Name
/// </summary>
public class PacketS2CZoneName : PacketS2C
{
    public const byte ID = 0x04;

    public PacketS2CZoneName(byte controllerID, byte zoneID, string name)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        WriteNullTerminatedString(name);
    }

    public override byte GetID() => ID;
}

File: src_js/server/packets/PacketS2CZoneName.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x04
 * Zone Name
 * Sends a zone's name
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unisgned Char) Zone ID
 *      (String) Name
 */
class PacketS2CZoneName extends PacketS2C {
    constructor(ctrlrID, zoneID, name) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeStringNT(name);
    }

    getID() {
        return 0x04;
    }
}

module.exports = PacketS2CZoneName;
